src/testdir/test_signals.vim

```diff
@@ -75,3 +77,37 @@ func Test_signal_PWR()
   bwipe!
   set updatetime& updatecount&
 endfunc
+
+" Test signal INT. Handler sets got_int. It should be like typing CTRL-C.
+func Test_signal_INT()
+  if !HasSignal('INT')
+    throw 'Skipped: INT signal not supported'
+  endif
+
+  " Skip the rest of the test when running with valgrind as signal INT is not
+  " received somehow by Vim when running with valgrind.
+  let cmd = GetVimCommand()
+  if cmd =~ 'valgrind'
+    throw 'Skipped: cannot test signal INT with valgrind'
+  endif
+
+  if !CanRunVimInTerminal()
+    throw 'Skipped: cannot run vim in terminal'
+  endif
+  let buf = RunVimInTerminal('', {'rows': 6})
+
+  " Get the PID of Vim running in terminal.
+  call term_sendkeys(buf, ":echo '<' .. getpid() .. '>'\n")
+  call WaitForAssert({-> assert_match('^<\d\+>', term_getline(buf, 6))})
+  let pid = substitute(term_getline(buf, 6), '^<\(\d\+\)>.*', '\=submatch(1)', '')
+
+  " Check that an endless loop in Vim is interrupted by signal INT.
+  call term_sendkeys(buf, ":while 1 | endwhile\n")
+  call WaitForAssert({-> assert_equal(':while 1 | endwhile', term_getline(buf, 6))})
+  exe 'silent !kill -s INT ' .. pid
```

Comment:
You can also use job_stop() with "int" to send SIGINT to the Vim instance.

Reply:
I could not make it work with `job_stop(job, 'int')`.
The goal of this test to check handing of signal INT and not to test `job_stop()`.
So using the lower level `!kill -s INT pid` may be preferable anyway.